apps/src/music/views/Controls.tsx

```diff
@@ -128,11 +141,23 @@ const Controls: React.FunctionComponent<ControlsProps> = ({
           id="run-button"
           className={classNames(
             moduleStyles.runButton,
-            isLoading && moduleStyles.disabled
+            disableRun && moduleStyles.disabled
           )}
-          onClick={() => setPlaying(!isPlaying)}
+          onClick={() => {
+            // Only send a predict level report if this is a predict level and the predict
+            // answer was not locked.
+            if (isPredictLevel && !predictAnswerLocked) {
+              dispatch(
+                sendPredictLevelReport({
+                  appType: 'music',
+                  predictResponse: predictResponse,
+                })
+              );
+            }
```

Comment:
Should this be a lab2-common thunk? Can we reasonably assume that conditions to submit a predict response will be the same for all labs (`isPredictLevel && !predictAnswerLocked`)? I'm wary of having to copy this logic into each lab that supports predict levels, but at the same time wouldn't want to start introducing lab-specific exceptions into common lab2 code if we think this would diverge for any reason. Curious what you think @molly-moen

Reply:
That seems reasonable to me!